vizro-core/src/vizro/models/_navigation/accordion.py

```diff
@@ -35,8 +35,8 @@ def coerce_pages_type(cls, pages):
     @_log_call
     def build(self, *, active_page_id=None):
         # Hide navigation panel if there is only one page
-        if len(list(itertools.chain(*self.pages.values()))) == 1:
-            return html.Div(className="hidden")
+        if len(list(itertools.chain(*self.pages.values()))) == 1 and len(dash.page_registry.keys()) == MIN_NO_OF_PAGES:
+            return html.Div(className="hidden", id="nav_panel_outer")
```

Comment:
I find this a bit difficult to understand, because it's not immediately clear why `MIN_NO_OF_PAGES` is 2. I would just re-write the code a bit to be very explicit, so it's clearer that we always want the nav panel to disappear if there is only one page registered and in the accordion.

```suggestion
        num_registered_pages = len(dash.page_registry.keys() - {MODULE_PAGE_404})
        num_accordion_pages = len(list(itertools.chain(*self.pages.values())))
        if num_accordion_pages == num_registered_pages == 1:
            return html.Div(className="hidden", id="nav_panel_outer")
```

Reply:
Since we will treat each icon/navitem individually and agreed to remove the nav_controls_panel if there is one page and no controls then this additional check involving MIN_NO_OF_PAGES is no longer needed. 

So I will revert to only checking for `if len(list(itertools.chain(*self.pages.values()))) == 1:`